core-web/libs/data-access/src/lib/dot-style-classes/dot-style-classes.service.ts

```diff
@@ -0,0 +1,29 @@
+import { Observable } from 'rxjs';
+
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+
+export const STYLE_CLASSES_FILE_URL = '/application/templates/classes.json';
+
+/**
+ * @description This service fetchs the available style classes from dotcms users
+ *
+ * @export
+ * @class DotStyleClassesService
+ */
+@Injectable({
+    providedIn: 'root'
+})
+export class DotStyleClassesService {
+    constructor(private http: HttpClient) {}
+
+    /**
+     * @description This method fetchs the style classes from the classes.json file
+     *
+     * @return {*}
+     * @memberof DotStyleClassesService
+     */
+    getStyleClassesFromFile(): Observable<object> {
+        return this.http.get(STYLE_CLASSES_FILE_URL);
```

Comment:
what happen if the file does not exists, is it a fallback? fallback to default site, system site, hardcoded

Reply:
This is mostly because, sometimes when a request fails, we need to display pop ups or do custom actions, so that's why we want to handle the errors in the points where we use it